api.py

```diff
@@ -37,24 +86,26 @@ def get(self):
 		selected_semester = self.request.get('semester')
 		
 		if selected_semester:
-			members = Member.query(Member.show == True, Member.semesters_paid == selected_semester).order(Member.name).fetch(limit=None)
+			# Avoid throwing error 500 if a bad semester string is supplied
+			try:
+				selected_semester = float(selected_semester)
+			except ValueError:
+				self.error(400)
+				return
+			members = Member.query(Member.semesters_paid == selected_semester).order(Member.name).fetch(limit=None)
 		else:
-			members = Member.query(Member.show == True, Member.never_paid == False).order(Member.name).fetch(limit=None)
+			members = Member.query().order(Member.name).fetch(limit=None)
 		
-		output = []
-		for member in members:
-			output.append(format_member(member))
+		output = [format_member(member) for member in members]
 		
 		self.response.headers['Content-Type'] = 'application/json'
 		self.response.write(json.dumps(output))
 
 class MemberAPI(webapp2.RequestHandler):
-	def get(self):
+	def get(self, id):
```

Comment:
Why the move from URL parameter to part of the path?

Reply:
It is a lot cleaner to be part of the path instead of a URL parameter, makes the URL structure more readable, and also allows the ability to select only specific parts of the member object if desired. For instance, for performance reasons it may be desirable someday to just pull `/members/<id>/semestersPaid` for a large set of members, then this class can just have another optional parameter added with a case for that.

Reply:
When would you use `/members/<id>/semesterspaid` though?  The handling of that API call would be identical to the existing member API, except with `format_member` stripping out the extra fields.  Unless something changed since I last read through the GAE datastore docs, all datastore queries are the equivalent of `SELECT * FROM...`, so the load on the datastore would presumably be the same.

Reply:
Makes sense, code will be simpler if the members endpoint just handles everything. I will strike out MemberAPI and just add handling for `id` as a parameter to `/members`, and make it just return a list of one member if filtered by `id`